### PR TITLE
CE and AC button functionnal tests added as well

### PR DESCRIPTION
**Pull request by izconcept**
Fixes: #21 
### WHAT kind of change does this PR introduce?

* CE now clears the current number (entry)
* AC now clears the entire calculator

### HOW is this accomplished?

* The reset function is used to clear the current entry.

### Checklist
- [x] Docs have been added / updated to reflect the changes
- [x] I have reviewed and tested the changes :white_check_mark:
- [x] I have squashed my commits to have a reasonable amount of commits

### CODE REVIEW

File: app/src/test/java/com/simplemobiletools/calculator/MainActivityTest.kt [hunk 1]
```diff
@@ -3,6 +3,7 @@ package com.simplemobiletools.calculator
 import com.simplemobiletools.calculator.activities.MainActivity
 import com.simplemobiletools.calculator.helpers.*
 import junit.framework.Assert.assertEquals
+import kotlinx.android.synthetic.main.activity_main.*
```
Comment (jusleg) [inline app/src/test/java/com/simplemobiletools/calculator/MainActivityTest.kt]: what is this import doing?

File: app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java [hunk 2]
```diff
@@ -15,6 +15,7 @@
 import static android.support.test.espresso.assertion.ViewAssertions.matches;
 import static android.support.test.espresso.matcher.ViewMatchers.withId;
 import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static junit.framework.Assert.assertEquals;
```
Comment (jusleg) [inline app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java]: Don't think we need this import.

File: app/src/test/java/com/simplemobiletools/calculator/MainActivityTest.kt (hunk 1 above)
Comment (izconcept) [reply app/src/test/java/com/simplemobiletools/calculator/MainActivityTest.kt]: we need it to call things like `activity.btn_clear.text`

File: app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java (hunk 2 above)
Comment (izconcept) [reply app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java]: We don't. I'll remove it

Comment (almiche) [reply app/src/androidTest/java/com/simplemobiletools/calculator/MainActivityTest.java]: Nice catch 👮🏻‍♀️

File: app/src/main/kotlin/com/simplemobiletools/calculator/helpers/CalculatorImpl.kt [hunk 3]
```diff
@@ -127,6 +139,14 @@ class CalculatorImpl(calculator: Calculator, val context: Context) {
...
     }
 
+    private fun setClear(){
+       mCallback!!.setClear("CE")
+    }
+
+    private fun setAllClear(){
+        mCallback!!.setClear("AC")
+    }
+
```
Comment (almiche) [inline app/src/main/kotlin/com/simplemobiletools/calculator/helpers/CalculatorImpl.kt]: Very clever way of doing this. Almost inspired 🙏🏻

File: app/src/main/kotlin/com/simplemobiletools/calculator/helpers/CalculatorImpl.kt [hunk 4]
```diff
@@ -71,11 +72,21 @@ class CalculatorImpl(calculator: Calculator, val context: Context) {
...
+        if(!secondNumberSet){
+            handleReset()
+            setAllClear()
+        }
+        else {
+            firstNumber = 0.0
+            setAllClear()
+            setValue("0")
+            secondNumberSet = false
+        }
```
Comment (almiche) [inline app/src/main/kotlin/com/simplemobiletools/calculator/helpers/CalculatorImpl.kt]: Ouf ride the flags 🚩🎢

**Comment by almiche** [comment]
Excellent PR I approve of this.

**Pull request merged**